arbitrator/include/arbitrator_utils.hpp

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (C) 2019 LEIDOS.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+#ifndef __ARBITRATOR_UTILS_HPP__
+#define __ARBITRATOR_UTILS_HPP__
+
+#include <ros/ros.h>
+#include <cav_msgs/ManeuverPlan.h>
+
+/**
+ * \brief Macro definition to enable easier access to fields shared across the maneuver typees
+ * \param mvr The maneuver object to invoke the accessors on
+ * \param property The name of the field to access on the specific maneuver types. Must be shared by all extant maneuver types
+ * \return Expands to an expression (in the form of chained ternary operators) that evalutes to the desired field
+ */
+#define GET_MANEUVER_PROPERTY(mvr, property)\
```

Comment:
Maybe you can make it as a inline function instead of a Macro? In C++, the style guide says: "Avoid defining macros, especially in headers": https://google.github.io/styleguide/cppguide.html#Preprocessor_Macros

Reply:
For this one in particular I don't believe it can be anything other than a macro, due to the fact that it's actually doing something syntactically invalid for normal C++ code (using that property parameter to determine which field to access on an object by name). 

I can define it only in the CPP file if it makes a difference, but I'll have to refactor the other uses of it to call function that uses the macro instead.

Reply:
After discussion, we will keep this but add a TODO note for future enhancement.